functions/go/set-namespace/transformer/namespace.go

```diff
@@ -31,46 +30,51 @@ func SetNamespace(rl *fn.ResourceList) (bool, error) {
 		return true, nil
 	}
 	// Update "namespace" to the proper resources.
-	count := tc.Transform(rl.Items)
-	var result *fn.Result
-	if len(tc.Errors) != 0 {
-		errMsg := strings.Join(tc.Errors, "\n")
-		result = fn.GeneralResult(errMsg, fn.Error)
-	} else {
-		result = fn.GeneralResult(tc.LogResult(count), fn.Info)
-	}
-
+	oldNamespaces := map[string]struct{}{}
+	count := tc.Transform(rl.Items, oldNamespaces)
+	result := tc.LogResults(rl, count, oldNamespaces)
 	rl.Results = append(rl.Results, result)
 	return true, nil
 }
 
-func (p *NamespaceTransformer) LogResult(count int) string {
-	return fmt.Sprintf("namespace %q updated to %q, %d values changed", p.NamespaceMatcher, p.NewNamespace,
-		count)
+func (p *NamespaceTransformer) LogResults(rl *fn.ResourceList, count int, oldNamespaces map[string]struct{}) *fn.Result {
+	if len(p.Errors) != 0 {
+		errMsg := strings.Join(p.Errors, "\n")
+		return fn.GeneralResult(errMsg, fn.Error)
+	}
+	if dupErr := fn.CheckResourceDuplication(rl); dupErr != nil {
+		return fn.ErrorResult(dupErr)
+	}
+	if count == 0 {
+		msg := fmt.Sprintf("all namespaces are already %q. no value changed", p.NewNamespace)
+		return fn.GeneralResult(msg, fn.Info)
+	}
+
+	msg := fmt.Sprintf("namespace %q updated to %q, %d value(s) changed", reflect.ValueOf(oldNamespaces).MapKeys(),
```

Comment:
> reflect.ValueOf(oldNamespaces).MapKeys()

IMO it's not necessary to use `reflect` here.
We can simply convert the map to a slice.

Reply:
It will be something like:
```
var oldNsList []string
for oldNs := range oldNamespaces {
	oldNsList = append(oldNsList, oldNs)
}
fmt.Sprintf("message...", strings.Join(oldNsList), foo, bar)
```

It has a few more lines.But IMO it's easier to reason about than using `reflect`.

Reply:
done.